docs/tutorials/on-device-training/ios-app.md

```diff
@@ -0,0 +1,987 @@
+
+```ruby
+pod `onnxruntime-training-objc`, `~> 1.16.0`
+```
+
+Save the `Podfile` and run the following command to install the dependencies:
+
+```bash
+pod install
+```
+
+This will create a `MyVoice.xcworkspace` file in the project directory. Open the `xcworkspace` file in Xcode. This will open the project in Xcode with the dependencies installed.
+
+Now, create a new group in the project and name it `artifacts`. Drag and drop the artifacts generated in the previous section into the `artifacts` group. Make sure to select `Create folder references` and `Copy items if needed` options. This will add the artifacts to the project.
+
+Next, create a new group in the project and name it `recordings`. This group will contain the audio recordings that will be used for training. You can generate the recordings by running the [`recording_gen.py`](https://github.com/microsoft/onnxruntime-training-examples/blob/master/on_device_training/mobile/ios/recording_gen.py) script at the root of the project. Alternatively, you can also any other recordings of length 10 seconds in .wav format.
+The project structure should look like this:
+
+![Xcode Project Structure](../../../images/iOS_speaker_identification_screenshot_3.png)
+
+
+### Application Overview
+
+The application will consist of two main UI Views: `TrainingView` and `InferenceView`. The `TrainingView` is used to train the model on-device, and the `InferenceView` is used to perform inference with the trained model. Additionally, there is `ContentView` which is the home view of the application and contains buttons to navigate to the `TrainingView` and `InferenceView`.
+
+Additionally, we will also create `AudioRecorder` class to handle the recording of audio through the microphone. It will record 10 seconds of audio and output the audio data as a `Data` object, which can be used for training and inference purposes.
+
+We will have `Trainer` class which will handle the training and exporting of the model.
+
+Laslty, we will also create `VoiceIdentifier` class that will handle the inference with the trained model. 
+
+### Training the model
+
+First, we will create a `Trainer` class that will handle the training and exporting of the model. It will load the training artifacts, train the model on given audio, and export the trained model using onnnxruntime on device training APIs. The detailed documentation for the API can be found [here](../../../docs/api/objectivec/index.html).
+
+The `Trainer` class will have the following public methods:
+- `init()` - Initializes the training session and loads the training artifacts
+- `train(_ trainingData: [Data])` - Trains the model on the given user audio data. It will take in an array of `Data` objects, where each `Data` object represents the audio data of the user and use it along with some pre-recorded audio data to train the model.
+- `exportModelForInference()` - Exports the trained model for inference purposes
```

Comment:
```suggestion
- `init()` - Initializes the training session and loads the training artifacts.
- `train(_ trainingData: [Data])` - Trains the model on the given user audio data. It will take in an array of `Data` objects, where each `Data` object represents the audio data of the user, and use it along with some pre-recorded audio data to train the model.
- `exportModelForInference()` - Exports the trained model for inference purposes.
```